### PR TITLE
Add sub variant to nav component

### PR DESCRIPTION
**Pull request by dgalarza**
image

### CODE REVIEW

File: packages/components/nav/src/nav.tsx [hunk 1]
```diff
@@ -2,7 +2,15 @@ import styled from "styled-components";
 
 import { Flex, FlexProps } from "@buoysoftware/anchor-layout";
 
-export const Nav = styled((props) => {
+type Variant = "main" | "sub";
+
+interface OwnProps {
+  variant?: Variant;
```
Comment (acostanzo) [inline packages/components/nav/src/nav.tsx:8]: Is this the first introduction of a variant? If so, should we consider abandoning the term and make a separate component for SubNav?

Comment (dgalarza) [reply packages/components/nav/src/nav.tsx:8]: It is but I'm working on buttons which will likely have variants as well.

Comment (dgalarza) [reply packages/components/nav/src/nav.tsx:8]: It also feels like a good use for a variant here, where everything is the same except some minor styling differences. As opposed to our old text component which got to complicated with variants etc.

Comment (acostanzo) [reply packages/components/nav/src/nav.tsx:8]: Seems reasonable to me just wanted to make sure.

**Review by acostanzo — APPROVED** [review]

**Pull request merged**